cmake/modules/board_extensions.cmake

```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
we should not create modules just for the sake of creating modules.

Initial feeling is that the lookup for extensions probably belongs inside `boards.cmake`, and then the corresponding dts and kconfig modules decides how to use such information.

will look a little deeper into this and may come up with a small proposal on details.